### PR TITLE
Generator systematics

### PR DESCRIPTION
**Pull request by EmyrClement**
- Add generator weight systematics
- PDF and scaleup/down ME
- When producing unfolding files with generator weight, copy input file to worker node
- Add extra histogram to store info on phase space etc.

### CODE REVIEW

File: tools/Calculation.py [hunk 1]
```diff
@@ -113,9 +113,10 @@ def calculate_lower_and_upper_PDFuncertainty(central_measurement, pdf_uncertaint
     positive = []
     
     # split PDF uncertainties into downwards (negative) and upwards (positive) components
-    for index in range(1, 45):
+    for index in range(0, 100):
         pdf_weight = 'PDFWeights_%d' % index
         pdf_uncertainty = pdf_uncertainty_values[pdf_weight]
+        print index,central_measurement, pdf_uncertainty
         if index % 2 == 0:  # even == negative
```
Comment (kreczko) [inline tools/Calculation.py]: Is this still correct? The list used to start with an odd weight (positive) followed by the negative one.
The way it is currently done, the first weight will be counted as negative.
Could you please verify?

File: src/cross_section_measurement/05_make_tables.py [hunk 2]
```diff
@@ -677,11 +670,15 @@ def print_typical_systematics_table(central_values, errors, channel, toFile = Tr
...
 
+    pdf_uncertainties = ['PDF_total_lower', 'PDF_total_upper']
+
     ### # all MET uncertainties except JES as this is already included
     # met_uncertainties = [suffix for suffix in met_systematics if not 'JES' in suffix and not 'JER' in suffix]
     ### new_uncertainties = ['hadronisation', 'QCD_shape', 'PDF_total_lower', 'PDF_total_upper']
     rate_changing_systematics = measurement_config.rate_changing_systematics_names
     all_measurements = deepcopy(categories)
+    all_measurements.extend(pdf_uncertainties)
+    print all_measurements
```
Comment (kreczko) [inline src/cross_section_measurement/05_make_tables.py]: mark

File: tools/Calculation.py [hunk 3]
```diff
@@ -113,9 +113,10 @@ def calculate_lower_and_upper_PDFuncertainty(central_measurement, pdf_uncertaint
     positive = []
     
     # split PDF uncertainties into downwards (negative) and upwards (positive) components
-    for index in range(1, 45):
+    for index in range(0, 100):
         pdf_weight = 'PDFWeights_%d' % index
         pdf_uncertainty = pdf_uncertainty_values[pdf_weight]
+        print index,central_measurement, pdf_uncertainty
```
Comment (kreczko) [inline tools/Calculation.py]: Mark

File: experimental/BLTUnfold/produceUnfoldingHistograms.py [hunk 4]
```diff
@@ -121,7 +124,10 @@ def main():
     if int(options.centreOfMassEnergy) == 13:
         # file_name = fileNames['13TeV'][options.sample]
         file_name = getFileName('13TeV', options.sample, measurement_config)
-        print getFileName('13TeV', options.sample, measurement_config)
+        if options.generatorWeight >= 0:
+            file_name = 'localInputFile.root'
+        print file_name
```
Comment (kreczko) [inline experimental/BLTUnfold/produceUnfoldingHistograms.py]: Mark

**Comment by EmyrClement** [comment]
Print statements removed.

**Pull request merged**